### ISSUE TITLE
ogg-flac metadata is not recognized properly

### ISSUE THREAD

**Issue by mprevot** [Zeugma440/atldotnet#130]
## The problem

ogg-flac metadata is not recognized properly.

## Environment

* ATL version (or git revision) that exhibits the issue:
    <PackageReference Include="z440.atl.core" Version="4.3.0" />
* OS/version used to run ATL
    Windows 10 64bits, enterprise, VS2022, c# 10

## Details

After generating a putative ogg-flac file with  `ffmpeg.exe -i .\103_1.flac  -acodec flac -f ogg  103_1.flog` I could not get correct metadata with z440.atl.core 4.3.0

## Code To Reproduce Issue [ Good To Have ]

I used the class `Track` and few properties:

        public override string ToString() => $"Title: <{Title}>" +
            $"\nTitleAlternate: <{TitleAlternate}>" +
            $"\nPartName: <{PartName}>" +
            $"\nFilepath: <{Filepath}>" +
            $"\nDuration: {Duration}s" +
            $"\nDurationAlternate: {DurationAlternate}" +
            $"\nBitrate: {Bitrate}kbps" +
            $"\nSampleRate: {SampleRate}Hz" +
            $"\nAudioFormat: {AudioFormat.Name}" +
            $"\n{(IsVBR ? "variable" : "constant")} bitrate" +
            $"\n{(CodecFamily  == AudioDataIOFactory.CF_LOSSLESS ? "lossless" : "lossy")}" +
            $"\nCreationDate: <{CreationDate:D}>" +
            $"\nContentSizeBytes: {ContentSizeBytes*1e-6:N2}MB" +
            $"\nContentHex: <{string.Join("", ContentHex.Take(50))}>";

Result:
```
Title: <103_1>
TitleAlternate: <103_1>
PartName: <1>
Filepath: <D:\_\RecManagerTests\103_1.flog>
Duration: 0s
DurationAlternate: 00:00:00
Bitrate: 0kbps
SampleRate: 0Hz
AudioFormat: Unknown
constant bitrate
lossy
CreationDate: <lundi 3 janvier 2022>
ContentSizeBytes: 25,41MB
ContentHex: <0x4F67675300020000000000000000B32F772F00000000A913>
```

**Comment by Zeugma440** [comment]
Thanks for your feedback ! To be perfectly honest, I didn't know `flog` was a thing until today.

I'll do some tests and implement that soon.

**Comment by mprevot** [comment]
I made up the extension flog, however the association of flac codec and ogg container is described at https://xiph.org/flac/ogg_mapping.html and is implemented by few projects (vlc, ffmpeg at least).

**Comment by Zeugma440** [comment]
@mprevot OGG with embedded FLAC data can now be read using the latest version of `main`.

I'll release a new version of the library when you're happy with the results.

Please note that :
- Writing/updating metadata to this format is not supported yet. Tell me if you need it.
- The ffmpeg command line you provided creates an OGG file with embedded FLAC when the source FLAC file does not have any embedded picture. When it does, ffmpeg creates a Theora file instead. However, ATL doesn't support Theora yet. Tell me if you need it.
- The library won't recognize the `.flog` extension. You'll have to rename these files to `.ogg` or load them using a stream and the `audio/ogg` mime-type

```cs
using (FileStream fs = new FileStream(testFileLocation, FileMode.Open, FileAccess.Read, FileShare.Read))
{
    Track theTrack = new Track(fs, "audio/ogg");
    // do stuff there
}
```

Last but not least, if you don't want to test by building ATL locally, please tell me if you prefer to get a preview DLL.

Cheers~

**Comment by mprevot** [comment]
Sounds really great. I'll give it a try. What about releasing a preview version on nuget platform to help testing new features ? We can stick to stable releases as well as opt-in to previews ;) It's much easier and faster on our side.

**Comment by Zeugma440** [comment]
Alright. I've just published [4.4.0-alpha](https://www.nuget.org/packages/z440.atl.core/4.4.0-alpha) on NuGet.

Any feedback is welcome.

**Comment by mprevot** [comment]
Having the alpha version is great. It works fine with few properties:

```
            $"\nDuration: {_track.Duration}s" +
            $"\nBitrate: {_track.Bitrate}kbps" +
            $"\nSampleRate: {_track.SampleRate}Hz" +
            $"\nAudioFormat: {_track.AudioFormat.Name}" +
            $"\n{(_track.IsVBR ? "variable" : "constant")} bitrate" +
            $"\n{(_track.CodecFamily == AudioDataIOFactory.CF_LOSSLESS ? "lossless" : "lossy")}";
```
            
Continuing tests...

**Comment by mprevot** [comment]
Having the alpha version is great. It works fine with few properties:

```
            $"\nDuration: {_track.Duration}s" +
            $"\nBitrate: {_track.Bitrate}kbps" +
            $"\nSampleRate: {_track.SampleRate}Hz" +
            $"\nAudioFormat: {_track.AudioFormat.Name}" +
            $"\n{(_track.IsVBR ? "variable" : "constant")} bitrate" +
            $"\n{(_track.CodecFamily == AudioDataIOFactory.CF_LOSSLESS ? "lossless" : "lossy")}";
```
            
Continuing tests...

**Comment by Zeugma440** [comment]
@mprevot Any news on that issue ? Can these changes be released, or do you need anything else ?

**Comment by Zeugma440** [comment]
Published in today's v4.04, as well as reading metadata from Theora files.

Please open a new issue if there's something missing.
